src/shared/qrcode-util.c

```diff
@@ -36,31 +36,47 @@ int dlopen_qrencode(void) {
         return r;
 }
 
-static void print_border(FILE *output, unsigned width) {
+static void print_positioned_border(FILE *output, unsigned width, unsigned row, unsigned column) {
         /* Four rows of border */
+        int fd = fileno(output);
+        if (fd < 0)
+                return (void)log_warning_errno(fd, "unable to get file descriptor from the file stream: %m");
+
+        if (row > 0 || column > 0)
+                set_terminal_cursor_position(fd, row, column);
+
         for (unsigned y = 0; y < 4; y += 2) {
                 fputs(ANSI_WHITE_ON_BLACK, output);
 
                 for (unsigned x = 0; x < 4 + width + 4; x++)
                         fputs(UNICODE_FULL_BLOCK, output);
 
                 fputs(ANSI_NORMAL "\n", output);
+                set_terminal_cursor_position(fd, row + 1, column);
```

Comment:
all of these new operations need to happen only if row or column are set, here and below

Reply:
> all of these new operations need to happen only if row or column are set, here and below

I think I'll split up to two functions: The normal ```print_qrcode``` that we call when no position is to be set, and the ```print_positioned_qrcode``` that gets called when the QR code's position is specified.